encrypted/hello_encrypted/hello_encrypted.c

```diff
@@ -15,10 +15,29 @@ int main() {
     stdio_init_all();
 
 #if PICO_CRT0_IMAGE_TYPE_TBYB
-    // If TBYB image, then buy it
-    uint8_t* buffer = malloc(4096);
-    rom_explicit_buy(buffer, 4096);
-    free(buffer);
+    boot_info_t boot_info = {};
+    int ret = rom_get_boot_info(&boot_info);
+    if (ret) {
+        // BOOT_TBYB_AND_UPDATE_FLAG_BUY_PENDING will always be set, but check anyway
+        if (boot_info.tbyb_and_update_info & BOOT_TBYB_AND_UPDATE_FLAG_BUY_PENDING) {
+            // Need to check flash_update_base is set to see if this is a TBYB update
+            uint32_t flash_update_base = boot_info.reboot_params[0];
+            if (flash_update_base) {
+                printf("Perform self-check... ");
+                if (1 == 1) {
+                    printf("passed\n");
```

Comment:
???

Reply:
Okay, perhaps it's worth adding a comment clarifying that then?

Reply:
Also, would it be safer / more idiomatic to move the `rom_explicit_buy` part to be _inside_ this "fake self-check" if-block?

Reply:
> Also, would it be safer / more idiomatic to move the `rom_explicit_buy` part to be _inside_ this "fake self-check" if-block?

No, you need to call `rom_explicit_buy` on every boot due to TBYB intricacies with self-decrypting binaries (which is why I put this sample code in to show what to do), but perform the self check beforehand if there's also a `flash_update_base` - see the picotool PR for an explanation of the intricacies, I put it in a comment somewhere